lib/ansible/modules/notification/mattermost.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Benjamin Jolivot <bjolivot@gmail.com>
+# Inspired by slack module :
+#    # (c) 2017, Steve Pletcher <steve@steve-pletcher.com>
+#    # (c) 2016, René Moser <mail@renemoser.net>
+#    # (c) 2015, Stefan Berggren <nsg@nsg.cc>
+#    # (c) 2014, Ramon de la Fuente <ramon@delafuente.nl>)
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = """
+module: mattermost
+short_description: Send Mattermost notifications
+description:
+    - The M(mattermost) module sends notifications to U(http://your.mattermost.url) via the Incoming WebHook integration
+version_added: "2.3"
+author: "Benjamin Jolivot (@bjolivot)"
+options:
+  url:
+    description:
+      - Mattermost url (i.e. http://mattermost.yourcompany.com)
+    required: true
+  api_key:
+    description:
+      - Mattermost webhook api key. Log into your mattermost site, go to
+        Menu -> Integration -> Incomming Webhook -> Add Incomming Webhook
+        This will give you full URL. api_key is the last part.
+        http://mattermost.example.com/hooks/C(API_KEY)
+    required: true
+  text:
+    description:
+      - Text to send. Note that the module does not handle escaping characters.
+    required: true
+  channel:
+    description:
+      - Channel to send the message to. If absent, the message goes to the channel selected for the I(api_key).
+    required: false
```

Comment:
I would drop the `required: false`. It is no longer needed to specify this for parameters that are not required.